### PR TITLE
Updated the management of stderr and stdout for child processes.

### PR DESCRIPTION
**Pull request by mww-aws**
*Issue #, if available:*

*Description of changes:*
Fixes "hang" bug when too much data in stderr from solver
Now polls process every 100ms to emit stdout/stderr messages.
Turns off buffering of stdout/stderr.
Should improve debuggability (real time emit of messages)

Also added a send_message utility, which is a command line tool for generating an SQS message for a solver.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: send_message [hunk 1]
```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env python3
+import argparse
+import logging
+import boto3
+import json       
+
```
Comment (eidelmanjonathan) [inline send_message:6]: Please do this in a separate PR

File: src/arg_satcomp_solver_base/solver/run_command.py [hunk 2]
```diff
@@ -18,21 +19,32 @@ def __init__(self, stdout_target_loc, stderr_target_loc):
...
+            if line != "":
+                self.logger.info(f"{str_name}: {line}")
+                file_handle.write(line)
+                return True    
+        return False
+            
     def run(self, cmd: list, output_directory: str):
-        self.logger.info("Running command: %s", str(cmd))
-        proc = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        os.environ['PYTHONUNBUFFERED'] = "1"
```
Comment (eidelmanjonathan) [inline src/arg_satcomp_solver_base/solver/run_command.py:32]: I don't think we want to set this any time run is called. Please move this so that it only runs once

File: src/arg_satcomp_solver_base/solver/run_command.py [hunk 3]
```diff
@@ -18,21 +19,32 @@ def __init__(self, stdout_target_loc, stderr_target_loc):
...
-                self.logger.info(f"STDERR: {line_str}")
-                stderr_handle.write(line_str)
-        return_code = proc.wait()
-
+            with open(os.path.join(output_directory, self.stderr_target_loc), "w") as stderr_handle:
+                self.logger.info("Running command: %s", str(cmd))
+                proc = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+                while True:
+                    return_code = proc.poll()
+                    while(self.process_line(proc.stdout, "STDOUT", stdout_handle)):
```
Comment (eidelmanjonathan) [inline src/arg_satcomp_solver_base/solver/run_command.py:40]: Won't this block until STDOUT is finished before writing any stderr?

**Review by eidelmanjonathan — CHANGES_REQUESTED** [review]

File: src/arg_satcomp_solver_base/solver/run_command.py (hunk 3 above)
Comment (mww-aws) [reply src/arg_satcomp_solver_base/solver/run_command.py:40]: It will block until all that has been written in the last second is complete.
